packages/response-ops/alerts_table/components/alerts_table.tsx

```diff
@@ -86,30 +78,30 @@ type AlertWithCaseIds = Alert & Required<Pick<Alert, typeof ALERT_CASE_IDS>>;
 type AlertWithMaintenanceWindowIds = Alert &
   Required<Pick<Alert, typeof ALERT_MAINTENANCE_WINDOW_IDS>>;
 
-const getCaseIdsFromAlerts = (alerts: Alerts) => [
+const getCaseIdsFromAlerts = (alerts: Alert[]) => [
   ...new Set(
     alerts
       .filter((alert): alert is AlertWithCaseIds => {
         const caseIds = alert[ALERT_CASE_IDS];
         return caseIds != null && caseIds.length > 0;
       })
-      .map((alert) => alert[ALERT_CASE_IDS])
+      .map((alert) => alert[ALERT_CASE_IDS] as string[])
```

Comment:
While a bit more verbose, these casts reflect more accurately the expected content of each specific field. This was previously untyped/reduced to string[], when in fact many more json types were allowed. This could be improved by adding a field extraction utility generic function or even better by generating/manually defining interfaces based on the alert indices mapping